test/specs/buffer-with-count.coffee

```diff
@@ -0,0 +1,107 @@
+{stream, prop, send, Kefir} = require('../test-helpers.coffee')
+
+describe 'bufferWithCount', ->
+
+  describe 'stream', ->
+
+    it 'should return stream', ->
+      expect(stream().bufferWithCount(1)).toBeStream()
+
+    it 'should activate/deactivate source', ->
+      a = stream()
+      expect(a.bufferWithCount(1)).toActivate(a)
+
+    it 'should be ended if source was ended', ->
+      expect(send(stream(), ['<end>']).bufferWithCount(1)).toEmit 
+      []
+      ['<end:current>']
```

Comment:
> i am confused about the { current } value. my understanding was that only properties have the notion of current values, but here we're passing in a stream?

It depends on what we consider "current" value. In tests `{ current: something }` or `'<end:current>'` only means that the callback was called immediately before subscribing method returns, i.e.:

```js
obs.onValue(() => { log(1) });
log(2);

// logs
// 1
// 2
```
 
See also http://rpominov.github.io/kefir/#current-in-streams